packages/react-scripts/config/jest/typescriptTransform.js

```diff
@@ -0,0 +1,89 @@
+// @remove-on-eject-begin
+// @remove-file-on-eject typescript
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const crypto = require('crypto');
+const path = require('path');
+const paths = require('../paths');
+const tsc = (function() {
+  try {
+    return require('react-scripts-plugin-typescript').tsc;
+  } catch (e) {
+    return require('typescript');
```

Comment:
This would only happen post eject, where throwing would happen if the user had malformed node_modules; by default this will "work". We could add a helpful message, I suppose.